From: Irfan Mohammad <irfan@graphiant.com>
Date: Sun, 11 Dec 2022 23:57:42 +0000
Subject: more descriptive debugs for no datastore changes

When there are no datastore changes in an sr_apply_changes call, it is
useful to know what caused the changes to vanish.

For example, was it validation, or an update from a subscriber.

Also reduce it to debug as it can be noisy at INFO level.
---
 src/sysrepo.c | 7 ++++---
 1 file changed, 4 insertions(+), 3 deletions(-)

diff --git a/src/sysrepo.c b/src/sysrepo.c
index 8ea6cda..2cb639e 100644
--- a/src/sysrepo.c
+++ b/src/sysrepo.c
@@ -3693,7 +3693,7 @@ sr_changes_notify_store(struct sr_mod_info_s *mod_info, sr_session_ctx_t *sessio
     }
 
     if (!mod_info->diff) {
-        SR_LOG_INF("No \"%s\" datastore changes to apply.", sr_ds2str(mod_info->ds));
+        SR_LOG_DBG("No \"%s\" datastore changes to apply.", sr_ds2str(mod_info->ds));
         goto store;
     }
 
@@ -3757,7 +3757,7 @@ sr_changes_notify_store(struct sr_mod_info_s *mod_info, sr_session_ctx_t *sessio
 
     if (!mod_info->diff) {
         /* diff can disappear after validation */
-        SR_LOG_INF("No \"%s\" datastore changes to apply.", sr_ds2str(mod_info->ds));
+        SR_LOG_DBG("No \"%s\" datastore changes to apply after validation.", sr_ds2str(mod_info->ds));
         goto store;
     }
 
@@ -3779,7 +3779,7 @@ sr_changes_notify_store(struct sr_mod_info_s *mod_info, sr_session_ctx_t *sessio
     }
 
     if (!mod_info->diff) {
-        SR_LOG_INF("No \"%s\" datastore changes to apply.", sr_ds2str(mod_info->ds));
+        SR_LOG_DBG("No \"%s\" datastore changes to apply after update.", sr_ds2str(mod_info->ds));
         goto store;
     }
 
@@ -3899,6 +3899,7 @@ cleanup:
         sr_errinfo_merge(&err_info, cb_err_info);
         sr_errinfo_new(&err_info, SR_ERR_CALLBACK_FAILED, "User callback failed.");
     }
+    SR_LOG_INF("%s for %s %s.", __func__, sr_ds2str(session->ds), err_info? "failed" : "success");
     return sr_api_ret(session, err_info);
 }
 
